tools/defs.bzl

```diff
@@ -25,6 +25,12 @@ def closure_js_library(**kwargs):
   for non-test JS source files.
   """
   kwargs.setdefault("convention", "GOOGLE")
+  suppress = kwargs.pop("suppress", [])
```

Comment:
I've tried this locally and realized this is coming from when a JS lib is compiled to a binary.
So, the suppress change can be removed completely but add the following to the `closure_js_binary` below. Instead of this:

```
kwargs.setdefault("defs",
    "--assume_function_wrapper",
    "--new_type_inf",
    "--rewrite_polyfills=false",
])
_closure_js_binary_alias(**kwargs)
```

Add these flags:

```
kwargs.setdefault("defs", [
    "--assume_function_wrapper",
    "--rewrite_polyfills=false",
    "--new_type_inf",
    "--jscomp_warning=newCheckTypes",
    "--jscomp_off=newCheckTypesExtraChecks",
    "--hide_warnings_for=closure/goog",
])
_closure_js_binary_alias(**kwargs)
```

Basically this list:

        "--jscomp_warning=newCheckTypes",
        "--jscomp_off=newCheckTypesExtraChecks",
        "--hide_warnings_for=closure/goog",

It's almost the same like you did with`suppress` but it only disables it for closure library, while keeping warnings on for our own code. 

Found it in https://github.com/google/closure-compiler/wiki/Using-NTI-(new-type-inference).
The `new_type_inf` is a new, supposedly better, type inference engine, but the closure library hasn't been all ported to it. So, they recommend using those additional flags until the closure library is updated.

Reply:
yeah new type doesn't play nice with ES6 + old closure.

I've done the fixes. PTAL.